### ISSUE TITLE
additional_copy_path doesn't seem to work correctly

### ISSUE THREAD

**Issue by gregsymons** [neillturner/kitchen-ansible#188]
I'm not 100% sure the extent of this issue yet, but when using `additional_copy_path` to copy a setup playbook to the host under test, I get a ENOENT:

```
E, [2016-07-07T11:22:27.810052 #88758] ERROR -- Kitchen: ------Exception-------
E, [2016-07-07T11:22:27.810093 #88758] ERROR -- Kitchen: Class: Kitchen::ActionFailed
E, [2016-07-07T11:22:27.810110 #88758] ERROR -- Kitchen: Message: Failed to complete #converge action: [No such file or directory @ rb_sysopen - /var/folders/0h/8f6sk7hx6x59f_myrv0fy60h0000gp/T/assemblies-schemas-ubuntu-1404-sandbox-20160707-88758-afbk3d/test/integration/setup.yml]
E, [2016-07-07T11:22:27.810120 #88758] ERROR -- Kitchen: ---Nested Exception---
E, [2016-07-07T11:22:27.810129 #88758] ERROR -- Kitchen: Class: Errno::ENOENT
E, [2016-07-07T11:22:27.810170 #88758] ERROR -- Kitchen: Message: No such file or directory @ rb_sysopen - /var/folders/0h/8f6sk7hx6x59f_myrv0fy60h0000gp/T/assemblies-schemas-ubuntu-1404-sandbox-20160707-88758-afbk3d/test/integration/setup.yml
E, [2016-07-07T11:22:27.810180 #88758] ERROR -- Kitchen: ------Backtrace-------
E, [2016-07-07T11:22:27.810189 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:1392:in `initialize'
E, [2016-07-07T11:22:27.810198 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:1392:in `open'
E, [2016-07-07T11:22:27.810206 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:1392:in `block in copy_file'
E, [2016-07-07T11:22:27.810215 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:1391:in `open'
E, [2016-07-07T11:22:27.810223 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:1391:in `copy_file'
E, [2016-07-07T11:22:27.810232 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:485:in `copy_file'
E, [2016-07-07T11:22:27.810241 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:402:in `block in cp'
E, [2016-07-07T11:22:27.810249 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:1570:in `block in fu_each_src_dest'
E, [2016-07-07T11:22:27.810258 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:1586:in `fu_each_src_dest0'
E, [2016-07-07T11:22:27.810270 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:1568:in `fu_each_src_dest'
E, [2016-07-07T11:22:27.810279 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:401:in `cp'
E, [2016-07-07T11:22:27.810287 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/kitchen-ansible-0.44.5/lib/kitchen/provisioner/ansible_playbook.rb:914:in `block (2 levels) in prepare_additional_copy_path'
E, [2016-07-07T11:22:27.810296 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/find.rb:48:in `block (2 levels) in find'
E, [2016-07-07T11:22:27.810305 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/find.rb:47:in `catch'
E, [2016-07-07T11:22:27.810313 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/find.rb:47:in `block in find'
E, [2016-07-07T11:22:27.810322 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/find.rb:42:in `each'
E, [2016-07-07T11:22:27.810330 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/find.rb:42:in `find'
E, [2016-07-07T11:22:27.810347 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/kitchen-ansible-0.44.5/lib/kitchen/provisioner/ansible_playbook.rb:908:in `block in prepare_additional_copy_path'
E, [2016-07-07T11:22:27.810357 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/kitchen-ansible-0.44.5/lib/kitchen/provisioner/ansible_playbook.rb:906:in `each'
E, [2016-07-07T11:22:27.810366 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/kitchen-ansible-0.44.5/lib/kitchen/provisioner/ansible_playbook.rb:906:in `prepare_additional_copy_path'
E, [2016-07-07T11:22:27.810375 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/kitchen-ansible-0.44.5/lib/kitchen/provisioner/ansible_playbook.rb:251:in `create_sandbox'
E, [2016-07-07T11:22:27.810391 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/test-kitchen-1.5.0/lib/kitchen/driver/ssh_base.rb:72:in `converge'
E, [2016-07-07T11:22:27.810400 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/test-kitchen-1.5.0/lib/kitchen/instance.rb:547:in `legacy_ssh_base_converge'
E, [2016-07-07T11:22:27.810410 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/test-kitchen-1.5.0/lib/kitchen/instance.rb:371:in `block in converge_action'
E, [2016-07-07T11:22:27.810419 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/test-kitchen-1.5.0/lib/kitchen/instance.rb:495:in `call'
E, [2016-07-07T11:22:27.810429 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/test-kitchen-1.5.0/lib/kitchen/instance.rb:495:in `synchronize_or_call'
E, [2016-07-07T11:22:27.810437 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/test-kitchen-1.5.0/lib/kitchen/instance.rb:460:in `block in action'
E, [2016-07-07T11:22:27.810445 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/benchmark.rb:288:in `measure'
E, [2016-07-07T11:22:27.810454 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/test-kitchen-1.5.0/lib/kitchen/instance.rb:459:in `action'
E, [2016-07-07T11:22:27.810462 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/test-kitchen-1.5.0/lib/kitchen/instance.rb:369:in `converge_action'
E, [2016-07-07T11:22:27.810489 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/test-kitchen-1.5.0/lib/kitchen/instance.rb:348:in `block in transition_to'
E, [2016-07-07T11:22:27.810500 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/test-kitchen-1.5.0/lib/kitchen/instance.rb:347:in `each'
E, [2016-07-07T11:22:27.810509 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/test-kitchen-1.5.0/lib/kitchen/instance.rb:347:in `transition_to'
E, [2016-07-07T11:22:27.810517 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/test-kitchen-1.5.0/lib/kitchen/instance.rb:160:in `verify'
E, [2016-07-07T11:22:27.810526 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/test-kitchen-1.5.0/lib/kitchen/instance.rb:189:in `block in test'
E, [2016-07-07T11:22:27.810534 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/benchmark.rb:288:in `measure'
E, [2016-07-07T11:22:27.810542 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/test-kitchen-1.5.0/lib/kitchen/instance.rb:185:in `test'
E, [2016-07-07T11:22:27.810551 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/test-kitchen-1.5.0/lib/kitchen/command.rb:176:in `public_send'
E, [2016-07-07T11:22:27.810559 #88758] ERROR -- Kitchen: /Users/gregsymons/.rvm/gems/ruby-2.2.1@u2you/gems/test-kitchen-1.5.0/lib/kitchen/command.rb:176:in `block (2 levels) in run_action'
E, [2016-07-07T11:22:27.810567 #88758] ERROR -- Kitchen: ----------------------
```

The relevant snippet of my .kitchen.yml:

```yaml
provisioner:
  additional_copy_path:
    - test/integration/setup.yml
```

This works in 0.44.2

**Comment by lobsterdore** [comment]
Not sure if this is related,but I am seeing broken additional copy path behaviour as well, the culprit is this (change)[https://github.com/neillturner/kitchen-ansible/commit/f025dce372e6e760d1c27c75b5d1f396d0325cd5]

Notice how:

destination = File.join(sandbox_path, File.basename(file))

Has changed to:

destination = File.join(sandbox_path, files)

When the role path is set it uses File.basename for the additional copy path values, so this change breaks the role path.

**Comment by nmische** [comment]
I'm running into the same issue as @lobsterdore. I would prefer the previous behavior:

```
destination = File.join(sandbox_path, File.basename(file))
```

**Comment by nmische** [comment]
Looking a bit more at the code, I'm not sure how this would work given the `Find` approach used in `0.44.x`. Version `0.43.0` seems to work for me.

**Comment by neillturner** [comment]
I will restore the old behaviour and make the new behaviour available by a different parameter recursive_additional_copy_path. Will release tomorrow.

**Comment by nmische** [comment]
I _think_ you may need to do the latter as I don't see an easy way to change `destination` back to `File.join(sandbox_path, File.basename(file))` given the `Find` logic.

**Comment by neillturner** [comment]
fixed in 0.44.6. reverted additional_copy_path code and added recursive_additional_copy_path option
